src/main/java/com/notably/MainApp.java

```diff
@@ -86,22 +84,22 @@ public void init() throws Exception {
      */
     private Model initModelManager(Storage storage, BlockModel blockModel, SuggestionModel suggestionModel,
         ViewStateModel viewStateModel, ReadOnlyUserPrefModel userPrefs) {
-        Optional<BlockTree> blockTreeOptional;
-        BlockTree initialData;
+        Optional<BlockModel> blockModelOptional;
+        BlockModel initialData;
         try {
-            blockTreeOptional = storage.readBlockTree();
-            if (!blockTreeOptional.isPresent()) {
+            blockModelOptional = storage.readBlockModel();
+            if (!blockModelOptional.isPresent()) {
                 logger.info("Data file not found. Will be starting with a sample BlockTree");
             }
-            initialData = blockTreeOptional.orElseGet(SampleDataUtil::getSampleBlockTree);
+            initialData = blockModelOptional.orElseGet(SampleDataUtil::getSampleBlockModel);
         } catch (DataConversionException e) {
             logger.warning("Data file not in the correct format. Will be starting with an empty BlockTree");
-            initialData = new BlockTreeImpl();
+            initialData = new BlockModelImpl();
         } catch (IOException e) {
             logger.warning("Problem while reading from the file. Will be starting with an empty BlockTree");
-            initialData = new BlockTreeImpl();
+            initialData = new BlockModelImpl();
         }
-        blockModel.setBlockTree(initialData);
+        blockModel.resetData(initialData);
         return new ModelManager(blockModel, suggestionModel, viewStateModel , userPrefs);
     }
```

Comment:
Actually, @firzanarmani , would you consider making the `initModelManager` accept only `storage` and `userPrefs` as parameters? `blockModel`, `suggestionModel`, and `viewStateModel` can then be constructed inside the `initModelManager` function.

I feel like doing it this way makes the instantiation of `ModelManager` more straightforward. What do you think?

Reply:
Sounds good to me!